### PR TITLE
[Editor-flow] Update templates to be typed and have highlight

### PR DESCRIPTION
**Pull request by yavorsky**
I'm diving into a project bit by bit, but not sure I like having some template logic (which is basically UI) with functional logic.

As a bonus I created small vscode syntax extension, which understand start of template literal after ts expression and highlight it. It's not polished (not working if we are wrapping some stuff in `'`, but I's much easier to see the module you want to template with highlights).

Before
Screen Shot 2020-01-23 at 1 11 30 AM

After
Screen Shot 2020-01-23 at 1 06 28 AM

Moreover these templates are typed so it will be easier to use it on the functional side:
Screen Shot 2020-01-23 at 1 07 50 AM

PS I'm a bit sleepy after the plane and want to see any feedback / review if you have some time, but please don't merge. I want to verify test I added running on CI. (will do this morning)

And give 🖐to the plugin https://marketplace.visualstudio.com/items?itemName=yavorsky.vscode-ts-template-literal
😄😄😄

### CODE REVIEW

**Comment by amitdahan** [comment]
Very cool! I wonder if `Opts` can somehow be auto-inferred 🤔

File: packages/yoshi-flow-editor/.vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "recommendations": [
+    "yavorsky.vscode-ts-template-literal"
```
Comment (tuchk4) [inline packages/yoshi-flow-editor/.vscode/extensions.json]: 👮‍♂️

File: packages/create-yoshi-app/templates/flow-editor/typescript/{%packagejson%} [hunk 2]
```diff
@@ -54,8 +54,8 @@
     "i18next": "^11.6.0",
     "lodash": "^4.17.11",
     "prop-types": "~15.6.0",
-    "react": "^16.9.0",
-    "react-dom": "^16.9.0",
+    "react": "~16.6.3",
+    "react-dom": "~16.6.3",
```
Comment (yavorsky) [inline packages/create-yoshi-app/templates/flow-editor/typescript/{%packagejson%}]: https://github.com/wix-private/bolt/blob/master/bolt-main/package.json#L93-L94

File: packages/create-yoshi-app/templates/flow-editor/typescript/{%packagejson%} [hunk 3]
```diff
@@ -29,8 +29,8 @@
     "@types/i18next": "^8.4.5",
     "@types/jest": "^24.0.0",
     "@types/node": "^8.0.0",
-    "@types/react": "~16.9.0",
-    "@types/react-dom": "~16.9.0",
+    "@types/react": "16.9.15",
```
Comment (ranyitz) [inline packages/create-yoshi-app/templates/flow-editor/typescript/{%packagejson%}]: I worth moving the types as well to `16.6.3`

Comment (yavorsky) [reply packages/create-yoshi-app/templates/flow-editor/typescript/{%packagejson%}]: Types are not exact matching react version. For example, there is not 16.6.x versions for `@types/react` at all 👀

**Comment by yavorsky** [comment]
I'll merge it. Let's discuss types version here and I can open new PR to update it if we need.

**Pull request merged**